### PR TITLE
9182 header tests

### PR DESCRIPTION
**Pull request by rhianwatkins1**
# 9182 Header Tests
[TP9182](https://moneyadviceservice.tpondemand.com/entity/9182-fincap-header) - Header
[TP9232](https://moneyadviceservice.tpondemand.com/entity/9232-add-tests-for-header-front-end) - Tests task

This PR includes the tests for the header component, originally removed as adding dough config broke them.

### CODE REVIEW

**Comment by davidtrussler** [comment]
👍

File: spec/javascripts/fixtures/Header.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<div>
+    <header data-site-header="" data-dough-component="Header" data-dough-header-config='{"textClose":"Close Search"}' >
```
Comment (willhall88) [inline spec/javascripts/fixtures/Header.html]: indentation, needs to go back a couple of spaces

File: spec/javascripts/fixtures/Header.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      <div>
+        <div>
+          <a data-mobile-search-button="">
+            <span 
+              data-mobile-search-text="" 
+              data-mobile-search-text-close="Close Search"
+            >Open Search</span>
+          </a>
+        </div>
+
```
Comment (willhall88) [inline spec/javascripts/fixtures/Header.html]: you can remove this empty line?

File: spec/javascripts/tests/Header_spec.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    requirejs(
+      ['jquery', 'Header'],
+      function($, Header) {
+        fixture.load('Header.html');
+
+        self.$header = $(fixture.el).find('[data-dough-component="Header"]');
+        self.obj = new Header(self.$header);
+
+
```
Comment (willhall88) [inline spec/javascripts/tests/Header_spec.js]: maybe only one empty line?

File: spec/javascripts/fixtures/Header.html (hunk 1 above)
Comment (willhall88) [inline spec/javascripts/fixtures/Header.html]: indentation, can be moved back a couple of spaces

File: spec/javascripts/fixtures/Header.html (hunk 2 above)
Comment (willhall88) [inline spec/javascripts/fixtures/Header.html]: maybe remove this empty line?

File: spec/javascripts/tests/Header_spec.js (hunk 3 above)
Comment (willhall88) [inline spec/javascripts/tests/Header_spec.js]: maybe only one empty line?

**Pull request merged**